core/CliMulti/Process.php

```diff
@@ -242,7 +242,7 @@ private static function isProcFSMounted()
 
     public static function getListOfRunningProcesses()
     {
-        $processes = `ps ex 2>/dev/null`;
+        $processes = `ps x 2>/dev/null`;
```

Comment:
I wonder if we could do something like 

```php
$returnCode = null;
$processes = system(ps ex 2>/dev/null, $returnCode);
if ($returnCode != 0) {
     $processes = system(ps x 2>/dev/null, $returnCode);
}
```

? And same below?

Reply:
Running -ex on secure linux increases the size of the audit log.  It is a policy violation.  Perhaps run 'x' first?  Least privileged first?

Reply:
I guess we could do this? Though if `ps ex` fails for a temporary reason and `ps x` fails because the OS is wrong that could result in some strange error messages. Would be a bit of an edge case though. Probably better than trying to detect the correct system/command to use. (I guess we could also allow the command options to be specified through config or something.)

Reply:
@leftmostlane sorry for getting back to you so late. Any chance you could look at the approach above where we fall back to `ps x` if `ps ex` doesn't work?

Reply:
Running 'ex' on secure Linux increases the size of the audit log and is a policy violation. On shared systems this fills up the audit log with access violations.  Perhaps run 'x' first? Test for least privileged first then fall back to 'ex' if that does not work.  
```
$returnCode = null;
$processes = system(ps x 2>/dev/null, $returnCode);
if ($returnCode != 0) {
     $processes = system(ps ex 2>/dev/null, $returnCode);
}
```

Reply:
Actually I did some various tests on different OS and some more research and it should be fine to use only `x`.

I also went back the history (took a while) and noticed I had added `ps -e` about 7 years ago in https://github.com/matomo-org/matomo/commit/113e403a9e5a6ccab2762fc321382b540b51f8d9#diff-daeea28082bcbb1d6765684dc7f99fb2R92

It was initially `-e` as in "Select all processes.".

Then it became `ps -ex`.

And then it became `ps ex` in https://github.com/matomo-org/matomo/commit/ca84f227dc1de30fd96e76ab5e7f61d5d5adca87

Which I think changes the meaning from `select all processes` to `Show the environment after the command.`. However, we don't need the environment. I think it was meant to be changed to `ps -e x` if that makes any sense and I understand things correctly.

The referenced issue says
```
On some systems the command 'ps -e' will not return processes that are run via cron (since they are not attached to a terminal). This causes the ./console climulti:request command to abort early and return no data, which in turn causes ./console core:archive to fail.
```

Which caused the change from `ps -ex` to `ps ex`. Unfortunately the referenced forum link does no longer work.

There is also https://github.com/matomo-org/matomo/issues/7054 which suggests to use `ps -e`. 

I'm planning to merge this PR as is within the next 7 days or so. Hoping it also fixes things on solaris maybe.

Otherwise we'd need to do things like `php_uname ()` and adjust the command based on the OS.

If anyone has some further insights be great to let me know.

Reply:
Sounds great!